In [1]:
import pandas as pd

In [1]:
import numpy as np
import seaborn as sns
from tesser import fit
from tesser import util
from tesser import tasks

In [2]:

data_dir = "/home/rodrigo/Dropbox/tesser_successor/Data/"

In [3]:
induct_all = util.load_induct(data_dir)
struct_all = util.load_struct(data_dir)

In [5]:
struct_all.SubjNum.unique()

array([100, 101, 102, 103, 104, 105, 106, 108, 109, 110, 111, 112, 113,
       114, 115, 116, 117, 119, 120, 121, 122, 123, 124, 125, 126, 127,
       128, 129, 130, 131, 132, 133, 135, 136, 137, 138])

In [ ]:
fixed = {}
var_names = ['gamma', 'alpha', 'tau']
var_bounds = {'alpha': [0, 1], 'gamma': [0, 1], 'tau': [0, 10]}
fit.fit_induct(struct_all, induct_all, fixed, var_names, var_bounds,
               verbose=False)

In [6]:
fixed = {'alpha': 0.3}
var_names = ['gamma', 'tau']
var_bounds = {'gamma': [0, 1], 'tau': [0, 10]}
fit.fit_induct(struct_all, induct_all, fixed, var_names, var_bounds,
               verbose=False)


({'alpha': 0.3, 'gamma': 0.7775984763983996, 'tau': 0.24866146359841507},
 -1064.1423263900851)

In [4]:
high =tasks.induct_avg_split_high(tasks.induct_avg_all(data_dir))
hp = high.participant

In [10]:
struct_high_all = struct_all.copy()
s_all = struct_high_all[struct_high_all['SubjNum'].isin(list(hp))]


In [16]:
induct_high_all = induct_all.copy()
i_all = induct_high_all[induct_high_all['SubjNum'].isin(list(hp))]

In [17]:
# High performers

fixed = {}
var_names = ['gamma', 'alpha', 'tau']
var_bounds = {'alpha': [0, 1], 'gamma': [0, 1], 'tau': [0, 10]}
fit.fit_induct(s_all, i_all, fixed, var_names, var_bounds)

({'gamma': 0.9066399717654446,
  'alpha': 0.05051122724126701,
  'tau': 0.6440759946851158},
 -440.5416981786939)

High performers as a group have a much lower likelihood -440.54


In [18]:
# low performers
low =tasks.induct_avg_split_low(tasks.induct_avg_all(data_dir))
lp = low.participant

In [19]:

struct_low_all = struct_all.copy()
s_all = struct_low_all[struct_low_all['SubjNum'].isin(list(lp))]

In [20]:
induct_low_all = induct_all.copy()
i_all = induct_low_all[induct_low_all['SubjNum'].isin(list(lp))]

In [21]:
# low performers logllikelihood 
fixed = {}
var_names = ['gamma', 'alpha', 'tau']
var_bounds = {'alpha': [0, 1], 'gamma': [0, 1], 'tau': [0, 10]}
fit.fit_induct(s_all, i_all, fixed, var_names, var_bounds)

({'gamma': 1.0, 'alpha': 0.5829999328907534, 'tau': 0.3244092381887595},
 -638.9331362159533)

Low performers as a group have a much lower likelihood


The sum individual log likelihood is -978.07139 which is much lower the group fitting 